--- libpreludedb-1.2.6/m4/libprelude.m4~	2015-08-04 01:25:50.741003746 +0800
+++ libpreludedb-1.2.6/m4/libprelude.m4	2015-09-11 12:34:56.651925859 +0800
@@ -50,6 +50,7 @@
     LIBPRELUDE_LDFLAGS=`$LIBPRELUDE_CONFIG $libprelude_config_args --ldflags`
     LIBPRELUDE_LIBS=`$LIBPRELUDE_CONFIG $libprelude_config_args --libs`
     LIBPRELUDE_PREFIX=`$LIBPRELUDE_CONFIG $libprelude_config_args --prefix`
+    LIBPRELUDE_CXX_LIBS=`$LIBPRELUDE_CONFIG $libprelude_config_args --libs --c++`
     LIBPRELUDE_CONFIG_PREFIX=`$LIBPRELUDE_CONFIG $libprelude_config_args --config-prefix`
     libprelude_config_version=`$LIBPRELUDE_CONFIG $libprelude_config_args --version`
 
@@ -177,6 +178,7 @@
   AC_SUBST(LIBPRELUDE_PTHREAD_CFLAGS)
   AC_SUBST(LIBPRELUDE_LDFLAGS)
   AC_SUBST(LIBPRELUDE_LIBS)
+  AC_SUBST(LIBPRELUDE_CXX_LIBS)
   AC_SUBST(LIBPRELUDE_PREFIX)
   AC_SUBST(LIBPRELUDE_CONFIG_PREFIX)
 ])
--- libpreludedb-1.2.6/bindings/c++/Makefile.am~	2015-08-04 01:25:50.695003748 +0800
+++ libpreludedb-1.2.6/bindings/c++/Makefile.am	2015-09-11 12:35:38.583074223 +0800
@@ -7,7 +7,7 @@
 libpreludedbcpp_la_SOURCES = preludedb.cxx preludedb-sql.cxx preludedb-error.cxx
 
 libpreludedbcpp_la_LDFLAGS = -no-undefined -version-info @LIBPRELUDEDBCPP_SONAME@
-libpreludedbcpp_la_LIBADD  = $(top_builddir)/src/.libs/libpreludedb.la
+libpreludedbcpp_la_LIBADD  = $(top_builddir)/src/.libs/libpreludedb.la @LIBPRELUDE_CXX_LIBS@
 libpreludedbcpp_la_CPPFLAGS = -I$(top_builddir) -I$(top_builddir)/src/include -I$(top_srcdir)/src/include -I$(top_builddir)/bindings/c++/include -I$(top_srcdir)/bindings/c++/include @LIBPRELUDE_CFLAGS@
 
 endif
